jms/org.wso2.carbon.transport.jms/src/test/java/org/wso2/carbon/transport/jms/test/QueueTopicAutoAckListeningTestCase.java

```diff
@@ -0,0 +1,126 @@
+        jmsServer = new JMSServer();
+        jmsServer.startServer();
+
+        // Create a queue transport listener
+        jmsQueueTransportListener = new JMSServerConnector("1");
+        queueTestMessageProcessor = new TestMessageProcessor();
+        jmsQueueTransportListener.setMessageProcessor(queueTestMessageProcessor);
+
+        // Create a topic transport listener
+        jmsTopicTransportListener = new JMSServerConnector("2");
+        topicTestMessageProcessor = new TestMessageProcessor();
+        jmsTopicTransportListener.setMessageProcessor(topicTestMessageProcessor);
+    }
+
+    /**
+     * JMS Server connector will start listening to a queue in auto ack mode and there will be a publisher publishing to
+     * the queue. After publishing the messages to queue, check whether the count of messages received is equal to
+     * number of messages sent.
+     */
+    @Test(groups = "jmsListening", description = "Testing whether queue listening is working correctly without any "
+            + "exceptions in auto ack mode")
+    public void queueListeningTestCase() {
+        try {
+            jmsQueueTransportListener.start(queueListeningParameters);
+            logger.info("JMS Transport Listener is starting to listen to the queue " + JMSTestConstants.QUEUE_NAME);
+            jmsServer.publishMessagesToQueue(JMSTestConstants.QUEUE_NAME);
+            Assert.assertEquals(queueTestMessageProcessor.getCount(), 10,
+                    "Expected message count is not received when " + "listening to queue "
+                            + JMSTestConstants.QUEUE_NAME);
+            jmsQueueTransportListener.stop();
+        } catch (Exception e) {
+            Assert.fail("Error while listening to queue");
+        }
+    }
+
+    /**
+     * JMS Server connector will start listening to a topic in auto ack mode and there will be a publisher publishing
+     * to the queue.After publishing the messages to queue, check whether the count of messages received is equal to
+     * number of messages sent.
+     */
+    @Test(groups = "jmsListening", description = "Testing whether topic listening is working correctly without any "
+            + "exceptions in auto ack mode")
+    public void topicListeningTestCase() {
+        try {
+            jmsTopicTransportListener.start(topicListeningParameters);
+            logger.info("JMS Transport Listener is starting to listen to the topic " + JMSTestConstants.TOPIC_NAME);
+            jmsServer.publishMessagesToTopic(JMSTestConstants.TOPIC_NAME);
+            Assert.assertEquals(topicTestMessageProcessor.getCount(), 10,
+                    "Expected message count is not received when " + "listening to topic "
+                            + JMSTestConstants.TOPIC_NAME);
+            jmsTopicTransportListener.stop();
+        } catch (Exception e) {
+            Assert.fail("Error while listening to topic");
```

Comment:
Simply throwing the error from the test case would fail it. We don't have to catch and Assert fail.